crates/stages/src/pipeline.rs

```diff
@@ -166,21 +183,23 @@ impl<DB: Database> Pipeline<DB> {
                 .await?;
 
             match next {
-                ControlFlow::Continue => {
-                    let tx = db.tx()?;
-                    previous_stage =
-                        Some((stage_id, stage_id.get_progress(&tx)?.unwrap_or_default()));
-                    tx.commit()?;
-                }
+                ControlFlow::NoProgress => {} // noop
+                ControlFlow::Continue { progress } => pipeline_progress.update(progress),
                 ControlFlow::Unwind { target, bad_block } => {
+                    // reset the sync state
+                    if let Some(ref updater) = self.sync_state_updater {
+                        updater.update_sync_state(SyncState::Downloading { target_block: target });
```

Comment:
Is this right? If our target is a block we've already downloaded should we be saying we're still downloading?

Reply:
the thing is that we don't know the target block for the headers stage until it's executed. for other stages we just use the progress of the previous stage (ultimately the headers stage)

Reply:
Mhm sounds good. We'll need to revisit all this before release anyway and document properly the difference between forward/reverse stage etc.